src/rebar_prv_dialyzer.erl

```diff
@@ -534,8 +542,7 @@ run_dialyzer(State, Opts, Output) ->
                      {timing, Timing} |
                      Opts],
             ?DEBUG("Running dialyzer with options: ~p~n", [Opts2]),
-            Warnings = format_warnings(rebar_state:opts(State),
-                                       Output, dialyzer:run(Opts2)),
+            Warnings = format_warnings(State, Output, dialyzer:run(Opts2)),
```

Comment:
You've already parsed the arguments for the output format and put them in `Opts` by then. You're now passing `output_format` to `dialyzer:run/1` and then ignoring the value and fetching it back again within `format_warnings/3`

I think there's some confusion about what the options are and how far deep they get passed.

At this point I believe that:
1. they should not be put in `Opts` as passed to Dialyzer
2. they could be fetched only within `format_warnings/3`
3. They can probably be found in the option dictionary returned by `rebar_state:opts/1` and could be done without changing the type signature of `format_warnings/3`

Reply:
Fixed 1 and 2. For 3, please see my reply on your other comment.